### PR TITLE
chore: added header checker

### LINKED ISSUE #1284: Add Header Check to android-maps-utils
We are currently not checking for copyright headers on android-maps-utils. Adding a GitHub action to check whether our files contain a copyright header will help us to keep all our files copyrighted.

### PR DESCRIPTION
**Pull request by kikoso**
This PR adds a new GitHub flow to check whether our files contain a copyright header.

---

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Edit the title of this pull request with a [semantic commit prefix](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#type) (e.g. "fix: "), which is necessary for automated release workflows to decide whether to generate a new release and what type it should be.
- [ ] Will this cause breaking changes to existing Java or Kotlin integrations? If so, ensure the commit has a `BREAKING CHANGE` footer so when this change is integrated a major version update is triggered. See: https://www.conventionalcommits.org/en/v1.0.0/
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #1284 🦕

### CODE REVIEW

**Comment by wangela** [comment]
Should we add .gradle files too?

**Comment by wangela** [comment]
This should be merged as a `chore:`

**Review by wangela — CHANGES_REQUESTED** [review]
Add .gradle

File: .github/header-checker-lint.yml
Comment (wangela) [inline .github/header-checker-lint.yml:1]: The header-check check is failing https://github.com/googlemaps/android-maps-utils/pull/1285/checks?check_run_id=20320966664

`.github/header-checker-lint.yml` should have a copyright year of 2024

**Review by wangela — CHANGES_REQUESTED** [review]

File: .github/header-checker-lint.yml
Comment (wangela) [reply .github/header-checker-lint.yml:1]: Sorry this is my fault for taking so long to review!

**Review by dkhawk — APPROVED** [review]

**Pull request merged**